agent/spec/lib/kontena/network_adapters/weave_spec.rb

```diff
@@ -47,6 +47,56 @@
     end
   end
 
+  describe '#config_changed?' do
+    let(:valid_image) do
+      "#{Kontena::NetworkAdapters::Weave::WEAVE_IMAGE}:#{Kontena::NetworkAdapters::Weave::WEAVE_VERSION}"
+    end
+
+    it 'returns false if config is the same' do
+      config = {
+        'grid' => {
+          'trusted_subnets' => []
+        }
+      }
+      weave_config = {
+        'Image' => valid_image,
+        'Cmd' => ['--trusted-subnets', '']
+      }
+      weave = spy(:weave, config: weave_config)
```

Comment:
Why `spy` instead of just `double`? Using `double` means that you get an rspec error if you forgot to mock something out, whereas `spy` just silently returns nil for everything..

Asking because I'm trying to fix `spy` -> `double` elsewhere :)